src/lib/components/data-encoder-status.svelte

```diff
@@ -6,16 +6,30 @@
     faLongArrowAltUp,
   } from '@fortawesome/free-solid-svg-icons';
 
+  import {
+    dataEncoderEndpoint,
+    lastDataEncoderStatus,
+  } from '$lib/stores/data-encoder-config';
+
   import {
     dataConverterPort,
     lastDataConverterStatus,
   } from '$lib/stores/data-converter-config';
 
   import Tooltip from '$lib/components/tooltip.svelte';
+  $: notRequested = $dataConverterPort
+    ? $lastDataConverterStatus === 'notRequested'
+    : $lastDataEncoderStatus === 'notRequested';
+  $: error = $dataConverterPort
+    ? $lastDataConverterStatus === 'error'
+    : $lastDataEncoderStatus === 'error';
+  $: success = $dataConverterPort
+    ? $lastDataConverterStatus === 'success'
+    : $lastDataEncoderStatus === 'success';
```

Comment:
Originally I had used booleans and then realized I was doing a lot of resets in the logic like if error success = false not requested = false error = true kind of stuff. But it may be more convenient if we moved back to that approach or used a derived store with the information involved.

Reply:
👍